### PR TITLE
Converts booking deletion to soft-delete + more robust cancellation

### PR DESCRIPTION
**Pull request by emrysal**
* Implements BookingStatus as I was working on it anyway; This converts cancellation into a soft-delete & we can proceed to deprecate the `confirmed` and `rejected` fields.

* Also makes booking attendees part more robust in the case that a booking exists without invitees (Not possible normally, but can indicate a system failure and this is better than crashing the page)

Status:
* **Pending**: Booking needs to be accepted or rejected by the organizer.
* **Accepted**: Organizer has accepted this booking
* **Rejected**: Organizer has rejected this booking
* **Cancelled**: Organizer or invitee has cancelled this booking

### CODE REVIEW

File: pages/api/cancel.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 import async from "async";
-import { deleteMeeting } from "../../lib/videoClient";
+import { deleteMeeting } from "@lib/videoClient";
+import { asStringOrNull } from "@lib/asStringOrNull";
+import { BookingStatus } from "@prisma/client";
 
 export default async function handler(req, res) {
   if (req.method == "POST") {
-    const uid = req.body.uid;
+    const uid: string | null = asStringOrNull(req.body.uid);
```
Comment (KATT) [inline pages/api/cancel.ts:10]:
```suggestion
    const uid = asStringOrNull(req.body.uid);
```

File: pages/api/cancel.ts [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
-    const bookingToDelete = await prisma.booking.findFirst({
+    const bookingToDelete = await prisma.booking.findUnique({
       where: {
-        uid: uid,
+        uid,
```
Comment (KATT) [inline pages/api/cancel.ts:14]: doesn't this cause a type error? 👀

**Review by KATT — COMMENTED** [review]
LGTM - the only reason I'm not approving is that i'm too unfamiliar with the area of the app to give proper sign-off

File: pages/api/cancel.ts (hunk 2 above)
Comment (emrysal) [reply pages/api/cancel.ts:14]: Doesn't seem to - should it? tested all flows through so not sure if it should've

**Review by pumfleet — APPROVED** [review]
Looks good to me :) Drop me a message before you merge so I can update the schema on the production database

**Review by PeerRich — APPROVED** [review]

**Comment by PeerRich** [comment]
> Looks good to me :) Drop me a message before you merge so I can update the schema on the production database

I think next time it's easier if you merge + update schema on your own :)

**Pull request merged**